test/integration/targets/apt/tasks/apt.yml

```
@@ -437,3 +437,48 @@
   file:
     path: /usr/sbin/policy-rc.d
     state: absent
+
+# https://github.com/ansible/ansible/issues/65325
+- name: install hello with apt (to test allow_change_held_packages option)
+  apt:
+    name: hello
+    state: present
+    allow_change_held_packages: yes
+  register: apt_result
+
+- name: check hello with dpkg
+  shell: dpkg-query -l hello
+  failed_when: False
+  register: dpkg_result
+
+- name: verify installation of hello
+  assert:
+    that:
+      - "apt_result.changed"
+      - "dpkg_result.rc == 0"
+
+- name: verify apt module outputs
+  assert:
+    that:
+      - "'changed' in apt_result"
```

These two tasks can be combined and should use `changed` test.
```suggestion
- name: verify installation of hello
  assert:
    that:
      - apt_result is changed
      - "dpkg_result.rc == 0"
```